File: examples/example_01_window.cpp

#include <loco/renderer/window/window.hpp>
#include <loco/utils/logging.hpp>

constexpr int DEFAULT_WIDTH = 1024;
constexpr int DEFAULT_HEIGHT = 768;

auto main() -> int {
    loco::renderer::WindowProperties properties;
    properties.backend = loco::renderer::eWindowBackend::TYPE_GLFW;
    properties.width = DEFAULT_WIDTH;
    properties.height = DEFAULT_HEIGHT;
    properties.title = "Example 01 - Window";

    auto window = std::make_unique<loco::renderer::Window>(properties);
    window->RegisterKeyboardCallback([](int key, int action) {
        LOG_TRACE("key: {0}, action: {1}", key, action);
    });

    while (window->active()) {
        window->Begin();
        // opengl-stuff can go here :)
        window->End();
    }

    return 0;
}

File: examples/python/example_01_window.py
#!/usr/bin/env python

import renderer as rdr

DEFAULT_WIDTH = 1024
DEFAULT_HEIGHT = 768


def main():
    config = rdr.WindowConfig()
    config.backend = rdr.WindowBackend.TYPE_GLFW
    config.width = DEFAULT_WIDTH
    config.height = DEFAULT_HEIGHT
    config.title = "Example 01 - Window"

    window = rdr.Window(config)

    def keycallback(key, action, mods):
        print("key: {}, action: {}. mods: {}".format(key, action, mods))
        if key == rdr.Keys.KEY_ESCAPE and action == rdr.KeyAction.PRESSED:
            window.RequestClose()

    window.RegisterKeyboardCallback(keycallback)

    while window.active:
        window.Begin()
        # opengl-stuff can go here :)
        window.End()


if __name__ == "__main__":
    main()
